contrib/fast-hdfs-resource/src/main/java/org/apache/ambari/fast_hdfs_resource/Runner.java

```diff
@@ -127,6 +128,25 @@ public static void main(String[] args)
           }
 
           if (resource.getAction().equals("create")) {
+            if ("file".equals(resource.getType())) {
+              // Check if the file already exists with same timestamp
+              // or not
+              FileStatus status = dfs.getFileStatus(pathHadoop);
+              long fileLength = status.getLen();
+              File f = new File(resource.getSource());
+              if (null == f || !f.exists()) {
```

Comment:
These checks should precede `dfs.getFileStatus` call, since target file size is not important if the operation is being skipped due to non-existent source file.

Reply:
@adoroszlai  ,
 I have implemented the review comments .

Reply:
Thanks!